docs/flip/1777-default-dtype.md

```diff
@@ -0,0 +1,102 @@
+# FLIP: Default dtypes
+
+
+- Start Date: 2022-01-11
+- FLIP PR: [#1776](https://github.com/google/flax/pull/1776)
+- FLIP Issue: [#1777](https://github.com/google/flax/issues/1777)
+
+
+## Summary
+
+This FLIP proposes to replace the default dtype which is currently hard-coded to float32, and instead use the JAX type promotion results to derive a default dtype from the input and parameters of a layer.
```

Comment:
This is a bit confusing: it sounds like there is a "hard coded default dtype", but it's rather that we have a "dtype that defaults to float32" and the output is hard-coded to be cast into that dtype.

Also consider using something like `module.dtype` to make the difference with `jnp.dtype`.

Reply:
I agree, "hard-coded" makes it sound like the user can't change it.

Reply:
Before #1739, whose design agrees with what was proposed in this FLIP, you could not change the parameter dtype of  the parameters (if not by specifying a custom initialiser with the dtype you wanted). I believe that's what is meant here by hardcoded?

Reply:
rephrased

Reply:
rephrased